exporter/clients.go

```diff
@@ -103,30 +181,185 @@ func (e *Exporter) extractConnectedClientMetrics(ch chan<- prometheus.Metric, c
+				e.options.Namespace,
+				"connected_client_total_memory_consumed_bytes",
+				"Total memory consumed by a client in its various buffers",
+				clientBaseLabels,
+			)
+			e.registerConstMetricGauge(
+				ch, "connected_client_total_memory_consumed_bytes", float64(info.TotMem),
+				clientBaseLabelsValues...,
+			)
+
+			e.metricDescriptions["connected_client_created_at_timestamp"] = newMetricDescr(
+				e.options.Namespace,
+				"connected_client_created_at_timestamp",
+				"A connected client's creation timestamp",
+				clientBaseLabels,
+			)
+			e.registerConstMetricGauge(
+				ch, "connected_client_created_at_timestamp", float64(info.CreatedAt),
+				clientBaseLabelsValues...,
+			)
+
+			e.metricDescriptions["connected_client_idle_since_timestamp"] = newMetricDescr(
+				e.options.Namespace,
+				"connected_client_idle_since_timestamp",
+				"A connected client's idle since timestamp",
+				clientBaseLabels,
+			)
+			e.registerConstMetricGauge(
+				ch, "connected_client_idle_since_timestamp", float64(info.CreatedAt),
+				clientBaseLabelsValues...,
+			)
+
+			e.metricDescriptions["connected_client_channel_subscriptions_count"] = newMetricDescr(
+				e.options.Namespace,
+				"connected_client_channel_subscriptions_count",
+				"A connected client's number of channel subscriptions",
+				clientBaseLabels,
+			)
+			e.registerConstMetricGauge(
+				ch, "connected_client_channel_subscriptions_count", float64(info.Sub),
+				clientBaseLabelsValues...,
+			)
+
+			e.metricDescriptions["connected_client_pattern_matching_subscriptions_count"] = newMetricDescr(
+				e.options.Namespace,
+				"connected_client_pattern_matching_subscriptions_count",
+				"A connected client's number of pattern matching subscriptions",
+				clientBaseLabels,
+			)
+			e.registerConstMetricGauge(
+				ch, "connected_client_pattern_matching_subscriptions_count", float64(info.Psub),
+				clientBaseLabelsValues...,
+			)
+
+			if info.Ssub != -1 {
+				e.metricDescriptions["connected_client_shard_channel_subscriptions_count"] = newMetricDescr(
+					e.options.Namespace,
+					"connected_client_shard_channel_subscriptions_count",
+					"a connected client's number of shard channel subscriptions",
+					clientBaseLabels,
+				)
+				e.registerConstMetricGauge(
+					ch, "connected_client_shard_channel_subscriptions_count", float64(info.Ssub),
+					clientBaseLabelsValues...,
+				)
+			}
+			if info.Watch != -1 {
+				e.metricDescriptions["connected_client_shard_channel_watched_keys"] = newMetricDescr(
+					e.options.Namespace,
+					"connected_client_shard_channel_watched_keys",
+					"a connected client's number of keys it's currently watching",
+					clientBaseLabels,
+				)
+				e.registerConstMetricGauge(
+					ch, "connected_client_shard_channel_watched_keys", float64(info.Ssub),
+					clientBaseLabelsValues...,
+				)
+			}
+
+			e.metricDescriptions["connected_client_query_buffer_length"] = newMetricDescr(
```

Comment:
`qbuf` and `qbuf-free` are both bytes ([Redis source](https://github.com/redis/redis/blob/7.4.0/src/networking.c#L2878)); `sds` is a string buffer with a variable size length field.